gr-blocks/python/blocks/sigmf_sink_minimal.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+#
+# Copyright 2022 Free Software Foundation, Inc.
+#
+# This file is part of GNU Radio
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+#
+
+from gnuradio import gr
+from . import blocks_python as blocks
+import numpy as np
+import json
+
+
+class sigmf_sink_minimal(gr.hier_block2):
+    """
+    A partial implementation of the SigMF standard to allow saving SigMF files from GNU Radio.
+    It is expected that this block will be replaced with gr-sigmf in the near future, and thus
+    it is immediately being depreciated.
+    For more information on SigMF see https://github.com/gnuradio/SigMF
+
+    Args:
+        item_size: complex, float, or short is all that is supported at the moment
+        filename: filename, NOT including the extension, SigMF uses .sigmf-meta and .sigmf-data which will be automatically added to your filename
+        sample_rate: sample rate in Hz
+        center_freq: optional center Frequency in Hz
+        author: optional string used to record author of recording
+        description: optional string used to store whatever other info you want about the recording
+        hw_info: optional string used to record hardware used in making of recording, e.g. SDR type and antenna, or whether it's simulated data
+    """
+
+    def __init__(self, item_size, filename, sample_rate, center_freq, author, description, hw_info, is_complex):
+
+        # Input type, which is included in .sigmf-meta file
+        if item_size == 8:
+            datatype_str = 'cf32_le'
+        elif item_size == 4:
+            datatype_str = 'rf32_le'
+        elif item_size == 2 and is_complex:
+            datatype_str = 'ci16_le'
+        elif item_size == 2 and not is_complex:
+            datatype_str = 'ri16_le'
+        else:
+            raise ValueError
+
+        if '.sigmf' in filename:
+            filename = filename.rsplit('.', 1)[0]
+        gr.log.info('Generating '+filename +
+                    '.sigmf-meta, writing SigMF data to '+filename+'.sigmf-data')
```

Comment:
```suggestion
        gr.log.info(f'Generating {filename}.sigmf-meta, writing SigMF data to {filename}.sigmf-data')
```
At least for me, f-strings are more readable.